File: shared/src/commonMain/kotlin/com/miam/kmm_miam_sdk/miam_core/model/Sponsor.kt
package com.miam.kmm_miam_sdk.miam_core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Sponsors(
    val sponsors:  List<Sponsor>,
)

@Serializable
data class Sponsor(
    val id: Int,
    val attributes: SponsorAttributes,
)

@Serializable
data class SponsorAttributes(
    val name : String?,
    @SerialName("logo-url")
    val logoUrl: String?
)

File: MiamIOSFramework/MiamIOSFramework/Model/Sponsor.swift
//
//  Sponsor.swift
//  MiamIOSFramework
//
//  Created by Vincent Kergonna on 23/02/2023.
//  Copyright © 2023 Miam. All rights reserved.
//

import Foundation
import miamCore

extension Sponsor {
    public var name: String {
        return self.attributes?.name ?? ""
    }

    public var logoURL: URL? {
        guard let logoUrl = self.attributes?.logoUrl else {
            return nil
        }

        return URL(string: logoUrl)
    }
}
